### PR TITLE
feat(spanner): add PG.OID support

### PR DESCRIPTION
**Pull request by skuruppu**
The PG.OID type is an int64 value with a PG_OID type annotation, mostly used by pg_catalog tables. This PR adds support for representing PG.OID values in the C# client.

Tests are added for being able to query a PG.OID value.

User tables can't be created with PG.OID as a column type so no tests were added for DML or mutations. Since the currently available set of pg_catalog tables don't have PG.OID columns, we also don't test being able to read a PG.OID from a table.

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs:35]: nit: Can you move these to after line 41 and update the comment in line 39? I think it's clearer if we have all the types that are not mapped automatically together.

**Review by amanda-tarafa — APPROVED** [review]
Hi @skuruppu:
I've left a tiny comment, otherwise looks good.
Please wait till @jskeet can take a look as well in case I missed something.

**Review by jskeet — APPROVED** [review]
Please address the comment from Amanda, but otherwise this looks okay to me. All the changes you *have* made look fine - but I can't vouch for whether there are other places that have been missed. Can I assume you looked for "jsonb" as an example?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs
Comment (skuruppu) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs]: That makes sense. All done.

**Comment by skuruppu** [comment]
> Please address the comment from Amanda, but otherwise this looks okay to me. All the changes you _have_ made look fine - but I can't vouch for whether there are other places that have been missed. Can I assume you looked for "jsonb" as an example?

Yeah exactly. I found all the PG.JSONB PRs (#9085, #9243, #9492) and followed exactly what those did. I think PG.JSONB is the closest type to PG.OID.

What I skipped was adding the type to `apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/AllTypesTableFixturePostgreSql.cs` since you shouldn't be able to create a user table with this thype.

**Comment by skuruppu** [comment]
@amanda-tarafa and @jskeet  I'll let you merge this when you're ready.

**Pull request merged**
